src/main/java/christmas/controller/OrderController.java

```diff
@@ -0,0 +1,117 @@
+
+    public OrderController(final StartView startView,
+                           final InputView inputView,
+                           final OutputView outputView,
+                           final OrderService orderService) {
+        this.startView = startView;
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.orderService = orderService;
+    }
+
+    public void order() {
+        printStartMessages();
+
+        final LocalDate dateOfVisit = InputUtil.retryOnInvalidInput(
+                this::inputAndValidateDateOfVisit,
+                INVALID_DATE_RE_ENTER.getMessage()
+        );
+
+        final Items items = InputUtil.retryOnInvalidInput(
+                this::inputAndCreateOrderItems,
+                INVALID_ORDER_RE_ENTER.getMessage()
+        );
+
+        final Order order = orderService.order(dateOfVisit, items);
+
+        outputOrder(order);
+    }
+
+    private void printStartMessages() {
+        startView.printWelcomeMessage();
+        startView.printMenu();
+        startView.printPrecaution();
+    }
+
+    private LocalDate inputAndValidateDateOfVisit() {
+        final String inputDateOfVisit = inputView.inputExpectedDateOfVisit();
+        return DateOfVisit.toLocalDate(inputDateOfVisit);
+    }
+
+    private Items inputAndCreateOrderItems() {
+        final String inputItemsWithCounts = inputView.inputOrderItemsWithCounts();
+        return Items.create(inputItemsWithCounts);
+    }
+
+    private void outputOrder(final Order order) {
+        final LocalDate date = order.getDateOfVisit();
+        final Items items = order.getItems();
+        final Benefits benefits = order.getBenefits();
+        final Badge badge = order.getBadge();
+
+        outputPreviewTitle(date);
+        outputOrderItems(items);
+        outputOrderBenefits(items, benefits);
+        outputMemberBadge(badge);
+    }
+
+    private void outputPreviewTitle(final LocalDate date) {
+        final OrderDateDto orderDateDto = DtoMapper.toOrderDateDto(date.getDayOfMonth());
+        outputView.outputPreviewTitle(orderDateDto);
+    }
+
+    private void outputOrderItems(final Items items) {
+        final Money totalPrice = items.calcTotalPrice();
+        final OrderItemsDto orderItemsDto = DtoMapper.toOrderItemsDto(
+                items.toMapWithNameKey(),
+                totalPrice.getAmount()
+        );
+        outputView.outputOrderItems(orderItemsDto);
+    }
+
+    private void outputOrderBenefits(final Items items, final Benefits benefits) {
+        final Money trickeryDiscount = benefits.calcTrickeryDiscount();
+        final OrderBenefitsDto orderBenefitsDto = DtoMapper.toOrderBenefitsDto(
+                benefits.toGiftsMapWithNameKey(),
+                benefits.toDiscountsMapWithNameKey(),
+                items.calcTotalPrice().getAmount(),
+                trickeryDiscount.getAmount()
+        );
+        outputView.outputBenefits(orderBenefitsDto);
+    }
```

Comment:
toOrderBenefitsDto에 items와 benefits를 그대로 넘겨주면 더 좋을 것 같습니다.
dto의 책임이 외부로 노출되어 있네요.
생성자의 인자도 크게 줄어들어서 코드도 간결하게 만들 수 있습니다.

````suggestion
    private void outputOrderBenefits(final Items items, final Benefits benefits) {
        outputView.outputBenefits(DtoMapper.toOrderBenefitsDto(items, benefits));
    }
```

Reply:
아아닛....그러네요......... Mapper를 제대로 사용하지 못했네요.... 중요한 부분 집어주셔서 감사합니다 👍 👍 👍 👍
코드예시까지 보여주셔서 바로 이해됐습니다 👏👏
DTO랑 DTO Mapper를 이번에 2번째로 써보는데 제가 쓴 방식대로 쓰면 Mapper의 장점을 살릴 수 없군용...